src/components/Homepage/Homepage.jsx

```diff
@@ -0,0 +1,108 @@
+import React, { useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { useSelector, useDispatch } from 'react-redux';
+import { fetchStockData } from '../../redux/Stocks/StocksSlice';
+import './Homepage.css';
+
+const Homepage = () => {
+  const [searchItem, setSearchItem] = useState('');
+  const [errorMsg, setErrorMsg] = useState('');
+  const navigate = useNavigate();
+  const dispatch = useDispatch();
+  const stocks = useSelector((state) => state.stocks.stocks);
+  const loading = useSelector((state) => state.stocks.status === 'loading');
+  const value = -10;
+
+  useEffect(() => {
+    dispatch(fetchStockData());
+  }, [dispatch]);
+
+  const filteredStock = stocks.filter((data) => data.name.toLowerCase().includes(searchItem.toLowerCase()));
+
+  const handleSearchInputChange = (e) => {
+    const inputValue = e.target.value.trim();
+
+    if (inputValue) {
+      setSearchItem(inputValue);
+      setErrorMsg('');
+    } else {
+      setSearchItem('');
+      setErrorMsg('Please enter a name');
+    }
+  };
+
```

Comment:
- [ ] As the requirements state about the design template, kindly make sure your follow the same tenplate design by  applying the  [design guidelines](https://www.behance.net/gallery/31579789/Ballhead-App-\(Free-PSDs\)) bellow :
   * Typography: font face, size, and weight
   * Layout: composition and space between elements.
   
 Check the defferences here 👇 
 
 Your design | the Template
---|---
<img width="216" alt="image" src="https://github.com/emarkees/stocks/assets/65068771/d3c0ef85-5d8b-4e36-bb20-1b244ba014ae"> | <img width="193" alt="image" src="https://github.com/emarkees/stocks/assets/65068771/85077986-ce37-46fa-8d84-df9251b7b81f"> <img width="178" alt="image" src="https://github.com/emarkees/stocks/assets/65068771/ff27b2db-43a2-44ff-8fab-42ec8fbec511">
<img width="211" alt="image" src="https://github.com/emarkees/stocks/assets/65068771/e505a6fd-824f-4c2a-a089-148470d02f15"> | <img width="190" alt="image" src="https://github.com/emarkees/stocks/assets/65068771/ecd7797d-59b9-4b9a-a60d-6bb02ac82689"> <img width="178" alt="image" src="https://github.com/emarkees/stocks/assets/65068771/836f830d-70dc-4543-b0e2-a4d715f8bc5d">

Reply:
done